cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/DRepActivity.hs

```diff
@@ -0,0 +1,301 @@
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  -- First proposal (set activity to something feasible in the test)
+  dRepActivityBeforeFirstProp <- getDRepActivityValue execConfig
+  dRepActivityBeforeFirstProp === 100 -- This is the default value
+
+  let firstPropVotes :: [(String, Int)]
+      firstPropVotes = zip (concatMap (uncurry replicate) [(6, "yes"), (4, "no")]) [1..]
+  annotateShow firstPropVotes
+
+  fgaInfo@(firstGovernanceActionTxId, firstGovernanceActionIndex) <-
+    makeActivityChangeProposal execConfig epochStateView (File configurationFile) (File socketPath)
+                               sbe gov "proposal1" Nothing 4 wallet0
+
+  voteChangeProposal execConfig epochStateView sbe gov "vote1"
+                     firstGovernanceActionTxId firstGovernanceActionIndex firstPropVotes wallet0
+
+  (EpochNo epochAfterFirstProp) <- getCurrentEpochNo epochStateView sbe
+  H.note_ $ "Epoch after first prop: " <> show epochAfterFirstProp
+
+  void $ waitUntilEpoch (File configurationFile) (File socketPath) (EpochNo (epochAfterFirstProp + 2))
+  dRepActivityAfterFirstProp <- getDRepActivityValue execConfig
+
+  dRepActivityAfterFirstProp === 4 -- This is what we just set, should work
+
+  result1 <- checkDRepState sbe (File configurationFile) (File socketPath) execConfig
+                            (Just . map (drepExpiry . snd) . Map.toList)
+  H.note_ $ "DRep expiration dates: " <> show result1
+
+  -- Second proposal (set activity to something else and it should fail, because of percentage being under 51)
+  let secondPropVotes :: [(String, Int)]
+      secondPropVotes = zip (concatMap (uncurry replicate) [(4, "yes")]) [1..]
+  annotateShow secondPropVotes
+
+  (secondGovernanceActionTxId, secondGovernanceActionIndex) <-
+    makeActivityChangeProposal execConfig epochStateView (File configurationFile) (File socketPath)
+                               sbe gov "proposal2" (Just fgaInfo) 7 wallet1
+
+  voteChangeProposal execConfig epochStateView sbe gov "vote2"
+                     secondGovernanceActionTxId secondGovernanceActionIndex secondPropVotes wallet1
+
+  (EpochNo epochAfterSecondProp) <- getCurrentEpochNo epochStateView sbe
+  H.note_ $ "Epoch after second prop: " <> show epochAfterSecondProp
+
+  void $ waitUntilEpoch (File configurationFile) (File socketPath) (EpochNo (epochAfterSecondProp + 5))
+  dRepActivityAfterSecondProp <- getDRepActivityValue execConfig
+
+  dRepActivityAfterSecondProp === 4 -- This is what we set in first prop, second prop should fail
+
+  result2 <- checkDRepState sbe (File configurationFile) (File socketPath) execConfig
+                            (Just . map (drepExpiry . snd) . Map.toList)
+  H.note_ $ "DRep expiration dates: " <> show result2
```

Comment:
Again here we should be asserting what the expected expiration epoch is supposed to be,